server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -273,9 +275,12 @@ public RemoteClusterStateService(
         this.remoteStateStats = new RemotePersistenceStats();
         this.namedWriteableRegistry = namedWriteableRegistry;
         this.indexMetadataUploadListeners = indexMetadataUploadListeners;
-        this.isPublicationEnabled = REMOTE_PUBLICATION_SETTING.get(settings)
-            && RemoteStoreNodeAttribute.isRemoteStoreClusterStateEnabled(settings)
-            && RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(settings);
+        this.isPublicationEnabled = new AtomicBoolean(
+            clusterSettings.get(REMOTE_PUBLICATION_SETTING)
+                && RemoteStoreNodeAttribute.isRemoteStoreClusterStateEnabled(settings)
+                && RemoteStoreNodeAttribute.isRemoteRoutingTableEnabled(settings)
```

Comment:
can these methods also be called configured instead of enabled?

Reply:
isRemoteStoreClusterStateEnabled should not be renamed as it is checks the value of REMOTE_CLUSTER_STATE_ENABLED_SETTING setting as well.
I have renamed isRemoteRoutingTableEnabled to isRemoteRoutingTableConfigured